src/rules/noUnusedVariableRule.ts

```diff
@@ -99,6 +99,19 @@ class NoUnusedVariablesWalker extends Lint.RuleWalker {
     public visitParameterDeclaration(node: ts.ParameterDeclaration): void {
         var variableName = node.name.text;
 
+        // Skip parameters in type literals. We don't want to generate
+        // warnings about unused variables in those.
+        if (node.parent && node.parent.symbol) {
```

Comment:
A better way of doing this than this should be to follow the format of the rest of this class -- for instance in visitIndexSignatureDeclaration, we turn skipParameterDeclaration on and off. So you should override visitTypeLiteral and turn the skipParameterDeclaration on and off from there, to make it consistent with the rest (and cleaner code, I think).